utils/eval/src/main/kotlin/dev/schlaubi/mikbot/eval/language/TypedExecutionResult.kt

```diff
@@ -0,0 +1,61 @@
+package dev.schlaubi.mikbot.eval.language
+
+import dev.kord.rest.builder.message.EmbedBuilder
+import dev.schlaubi.mikbot.eval.secrets.SecretExtensionPoint
+import dev.schlaubi.mikbot.haste.HasteClient
+import dev.schlaubi.mikbot.plugin.api.getExtensions
+import dev.schlaubi.mikbot.plugin.api.pluginSystem
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+private val extensions = pluginSystem.getExtensions<SecretExtensionPoint>()
+
+sealed class TypedExecutionResult : ExecutionResult() {
+    class Success(result: Any, val type: String) : TypedExecutionResult(), KoinComponent {
+        override val wasSuccessful: Boolean = true
+
+        private val hasteClient by inject<HasteClient>()
+
+        private val result =
+            extensions.asSequence()
+                .flatMap(SecretExtensionPoint::provideSecrets)
+                .filter(String::isNotBlank)
+                .fold(result.toString()) { acc, secret ->
+                    acc.replace(secret, "**REDACTED**")
+                }
+
+        override suspend fun EmbedBuilder.applyToEmbed() {
+            field("Result") {
+                if (result.length > 1024) {
```

Comment:
According to this, this is not 4096: https://discordjs.guide/popular-topics/embeds.html#embed-limits

Reply:
> A field's name is limited to 256 characters and its value to 1024 characters